string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,84 @@
+/* exported titleCase */
+
+/*
+- create an array of minor words
+- create variable for correct JavaScript capitalized
+- create variable for lower case JavaScript for condition
+- create variable for correct API capitalized
+- create variable for lower case API for condition
+- need to find out if individual word ends with colon so the next adjacent word will always get capitalized
+------------------
+- assuming input title letters are all lower case -- wrong, need to lowercase input sentence
+- Split input title with space to into a new array
+*/
+
+var minorWords = ['and', 'or', 'nor', 'but', 'a', 'an', 'the', 'as', 'at', 'by', 'for', 'in', 'of', 'on', 'per', 'to'];
+
+function titleCase(title) {
+  var titleArray = title.toLowerCase().split(' ');
+
+  var isMinorWordIndexes = [];
+  var isWordWithColonIndexes = [];
+  var isJavaScriptColon = [];
+  var isHyphenWordIndexes = [];
+  for (var i = 0; i < titleArray.length; i++) {
+    // check for minor word and append index at which minor is found
+    if (Array.prototype.includes.call(minorWords, titleArray[i])) {
+      isMinorWordIndexes.push(i);
+    }
+    // check for colon (:) and append index at which colon is found + 1 to mark the next adjacent word required capitalized
+    if (titleArray[i].includes(':')) {
+      isWordWithColonIndexes.push(i + 1);
+    }
+    // check if word is javascript: and append index at which javascript is found
+    if (titleArray[i] === 'javascript:') {
+      isJavaScriptColon.push(i);
+    }
+    // check if word has hyphen (-) and append index at which hypen-word is found
+    if (titleArray[i].includes('-')) {
+      isHyphenWordIndexes.push(i);
+    }
+  }
+
+  // first word ALWAYS get capitalized unless it's JavaScript:
+  var titledSentence = '';
+  if (!Array.prototype.includes.call(isJavaScriptColon, 0)) {
+    titledSentence += titleArray[0][0].toUpperCase() + titleArray[0].slice(1);
+  } else {
+    titledSentence += 'JavaScript:';
+  }
+
+  // Loop through the rest of the words in the array to concat to the final string
+  // starting at index 1 because index 0 is taken care of in step above
+  for (var j = 1; j < titleArray.length; j++) {
+    // need boolean to keep track whether to capitalize first letter of the current word in iteration
+    var yesCapitalize = true;
+    // checking if this index is in isMinorWordIndex
+    if (Array.prototype.includes.call(isMinorWordIndexes, j)) {
+      yesCapitalize = false;
+    }
+    // checking if this index need capitalize because previous index has colon
+    if (Array.prototype.includes.call(isWordWithColonIndexes, j)) {
```

Comment:
Again, no need for the `call` method here. I don't know where you picked this up, but it is _very rarely_ necessary